pkg/enclave_encrypt/client.go

```diff
@@ -0,0 +1,136 @@
+package enclave_encrypt
+
+import (
+	"crypto/ecdsa"
+	"crypto/elliptic"
+	"crypto/sha256"
+	"errors"
+	"fmt"
+	"reflect"
+
+	"github.com/btcsuite/btcutil/base58"
+	"github.com/cloudflare/circl/kem"
+)
+
+// An instance of the client side for enclave encrypt protocol. This should only be used for either
+// a SINGLE send or a single receive.
```

Comment:
we could "enforce" this by setting a private boolean value when it has been "used", but that doesn't necessarily prevent misuse 🤷